core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitUtils.java

```diff
@@ -183,26 +170,20 @@ public static boolean isTransactionalMetaColumn(String columnName, Set<String> a
    * @return whether the specified column is transactional or not
    */
   public static boolean isBeforeImageColumn(String columnName, TableMetadata tableMetadata) {
-    return isBeforeImageColumn(columnName, tableMetadata.getColumnNames());
-  }
-
-  /**
-   * Returns whether the specified column is a before image column or not.
-   *
-   * @param columnName a column name
-   * @param allColumnNames a set of all the column names
-   * @return whether the specified column is transactional or not
-   */
-  public static boolean isBeforeImageColumn(String columnName, Set<String> allColumnNames) {
-    if (!allColumnNames.contains(columnName)) {
+    if (!tableMetadata.getColumnNames().contains(columnName)
+        || tableMetadata.getPartitionKeyNames().contains(columnName)
+        || tableMetadata.getClusteringKeyNames().contains(columnName)) {
```

Comment:
I also added a condition where primary key columns are not before image columns.